### ISSUE TITLE
Cannot build test example (NodePath.split error)

### ISSUE THREAD

**Issue by rinfz** [godot-d/godot-d#82]
running `dub build` after pregenerate commands yields the following error:

```
..\..\..\AppData\Local\dub\packages\godot-d-0.1.12\godot-d\src\godot\core\nodepath.d(102,39): Error: `@nogc` function `godot.core.nodepath.NodePath.split` cannot call non-@nogc function `std.algorithm.searching.findSplitBefore!("a == b", wstring, OnlyResult!(immutable(wchar))).findSplitBefore`
..\..\..\AppData\Local\dub\packages\godot-d-0.1.12\godot-d\src\godot\core\nodepath.d(108,27): Error: none of the overloads of `this` are callable using argument types `(Result)`, candidates are:
..\..\..\AppData\Local\dub\packages\godot-d-0.1.12\godot-d\src\godot\core\string.d(69,17):        `godot.core.string.String.this(in godot_string str)`
..\..\..\AppData\Local\dub\packages\godot-d-0.1.12\godot-d\src\godot\core\string.d(77,2):        `__ctor(S)(in S str)`
  with `S = Result`
  must satisfy one of the following constraints:
`       isImplicitlyConvertible!(S, const(wchar_t)[])
       isImplicitlyConvertible!(S, const(wchar_t)*)`
..\..\..\AppData\Local\dub\packages\godot-d-0.1.12\godot-d\src\godot\core\string.d(96,2):        `__ctor(S)(in S str)`
  with `S = Result`
  must satisfy one of the following constraints:
`       isImplicitlyConvertible!(S, const(char)[])
       isImplicitlyConvertible!(S, const(char)*)`
C:\D\dmd2\windows\bin64\dmd.exe failed with exit code 1.
```

**Comment by Superbelko** [comment]
Quick fix is to update nodepath.d split method, though I haven't quite tested it yet

__src/godot/core/nodepath.d: around line 97__
```d
        NodePath[2] split() const
	{
		import std.string : indexOf  // 1: add this
		import std.range : only;
		static immutable wchar_t colon = ':';
		NodePath[2] ret;
		if(this == NodePath.init) return ret;
		String path = str();
		auto data = path.data();
		if(data[0] == colon)
		{
			ret[1] = this;
			return ret;
		}
		auto splitPos = data.indexOf(colon);  // 2: replace offending function
		if(splitPos == -1)  // 3: update condition
		{
			ret[0] = this;
			return ret;
		}
		ret[0] = NodePath(String(data[0..splitPos]));  // 4: update slicing
		ret[1] = NodePath(String(data[splitPos..$]));  // 4
		return ret;
	}
```


However I can't say if this fix is actually working since whenever I try using native script in new project it logs error on both editor/game start up
```
ERROR: _update_placeholder: Condition "!script_data" is true.
   At: modules/gdnative/nativescript/nativescript.cpp:92
```

**Comment by Superbelko** [comment]
Ah nevermind, it actually works now, my mistake was to put class name in "ScriptClass.Name" instead of actual "Class Name" property for .gdns asset.

**Comment by sheepandshepherd** [comment]
Thanks, that's a good solution. I've had trouble making anything with string decoding `nogc/nothrow` and didn't realize `indexOf` actually does work here. I ended up implementing something like `indexOf` using a nogc/nothrow UTF library, but need to take another look at those `std.string` functions.

The issue should be fixed in v0.1.13. It successfully compiled and ran with LDC+MSVC linker, but issue #43 is still preventing me from testing with DMD (and also breaking the CI for Windows builds).
